lib/comparable_basics.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2022, Google Inc. Please see the AUTHORS file for details.
+// All rights reserved. Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/// Utility extension methods for the [Comparable] class.
+extension ComparableBasics<T> on Comparable<T> {
+  /// Returns true if [this] should be ordered strictly before [other].
+  bool operator <(T other) => compareTo(other) < 0;
+
+  /// Returns true if [this] should be ordered strictly after [other].
+  bool operator >(T other) => compareTo(other) > 0;
+
+  /// Returns true if [this] should be ordered before or with [other].
+  bool operator <=(T other) => compareTo(other) <= 0;
+
+  /// Returns true if [this] should be ordered after or with [other].
+  bool operator >=(T other) => compareTo(other) >= 0;
+}
+
+/// Returns the greater of two [Comparable] objects.
```

Comment:
Can you update this to explicitly state the same method contract as `math.max` ([dartdoc]), specifically around NaN, 0.0, and equal arguments, and then add tests for the NaN and 0.0 cases? Ideally it should be the same for a user whichever `max` function they're actually calling. Same for `min`.

[dartdoc]: https://api.dart.dev/stable/2.16.2/dart-math/max.html

Reply:
Done.

Reply:
Thanks, just delegating to `math.max`/`min` definitely makes more sense. I still think we need to explicitly state though that for non-`num` arguments, it's unspecified which will be returned if they're equal. Can you add that to the dartdoc?

Reply:
Oops, fixed.